boms/integration-tests/src/test/java/com/google/cloud/MaximumLinkageErrorsTest.java

```diff
@@ -49,39 +47,22 @@ public void testForNewLinkageErrors()
     Path bomFile = Paths.get("../cloud-oss-bom/pom.xml");
     Bom bom = Bom.readBom(bomFile);
 
-    ImmutableSetMultimap<SymbolProblem, ClassFile> oldProblems =
-        LinkageChecker.create(baseline).findSymbolProblems();
+    ImmutableSet<LinkageProblem> oldProblems =
+        LinkageChecker.create(baseline).findLinkageProblems();
     LinkageChecker checker = LinkageChecker.create(bom);
-    ImmutableSetMultimap<SymbolProblem, ClassFile> currentProblems = checker.findSymbolProblems();
+    ImmutableSet<LinkageProblem> currentProblems = checker.findLinkageProblems();
 
     // This only tests for newly missing methods, not new references to
     // previously missing methods.
-    SetView<SymbolProblem> newProblems =
-        Sets.difference(currentProblems.keySet(), oldProblems.keySet());
+    SetView<LinkageProblem> newProblems = Sets.difference(currentProblems, oldProblems);
 
     // Check that no new linkage errors have been introduced since the baseline
     StringBuilder message = new StringBuilder("Baseline BOM: " + baselineCoordinates + "\n");
     if (!newProblems.isEmpty()) {
       message.append("Newly introduced problems:\n");
-      for (SymbolProblem problem : newProblems) {
-        message.append(problem + " referenced from " + currentProblems.get(problem) + "\n");
-      }
+      message.append(LinkageProblem.formatLinkageProblems(newProblems));
       Assert.fail(message.toString());
     }
-    
-    // If that passes, check whether there are any new references to missing methods:
-    for (SymbolProblem problem : currentProblems.keySet()) {
-      ImmutableSet<ClassFile> oldReferences = oldProblems.get(problem);
-      ImmutableSet<ClassFile> currentReferences = currentProblems.get(problem);
-      SetView<ClassFile> newReferences = Sets.difference(currentReferences, oldReferences);
-      if (!newReferences.isEmpty()) {
-        message.append("Newly introduced classes linking to " + problem + ":\n");
-        for (ClassFile classFile : newReferences) {
-          message.append("Link from " + classFile + "\n");
-        }
-        Assert.fail(message.toString());
-      }
-    }
```

Comment:
The check 2 lines above invalidate new references to existing symbol problems if any.

It was checking 2 types of errors

1. New linkage errors on symbols problem
1. New references to existing problems